src/lib/protocols/dhcp.c

```diff
@@ -114,6 +113,7 @@ void ndpi_search_dhcp_udp(struct ndpi_detection_module_struct *ndpi_struct,
 
             if(msg_type <= 8) {
               foundValidMsgType = 1;
+              flow->protos.dhcp.valid = 1;
```

Comment:
technically no it does the same thing as something like this would do flow->protos.dhcp.valid |= 0x01 would do because it happens before the other assignments. but i think replacing the above with flow->protos.dhcp.valid |= 0x01 makes the most sense. good point.